napari/components/viewer_model.py

```diff
@@ -68,41 +81,50 @@ class ViewerModel(KeymapProvider, MousemapProvider):
         Contains axes, indices, dimensions and sliders.
     """
 
+    # After https://github.com/samuelcolvin/pydantic/pull/2196 is released
+    # make all of these fields with allow_mutation=False
+    axes: Axes = Axes()
+    camera: Camera = Camera()
+    cursor: Cursor = Cursor()
+    dims: Dims = Dims()
+    grid: GridCanvas = GridCanvas()
+    layers: LayerList = Field(
+        default_factory=LayerList
+    )  # Need to create custom JSON encoder for layer!
+    scale_bar: ScaleBar = ScaleBar()
+
+    active_layer: Optional[
+        Layer
+    ] = None  # Would be nice to remove this/ make it layer name instead of layer?
+    help: str = ''
+    status: str = 'Ready'
+    theme: str = DEFAULT_THEME
```

Comment:
perhaps we should make a `Theme` enum that gets dynamically populated?
```
Themes = StringEnum('Themes', [(x.upper(), auto()) for x in available_themes()])
Themes.DEFAULT = Themes.DARK
```

Reply:
What if someone adds a theme during run time?

Reply:
good point.  ok, let's leave that for later.  Theme validation shouldn't be on the viewer model though...  but we can make a proper theme model later.

Reply:
yeah sounds good., i feel like with settings coming there could more improvements to how we handle this in the future, but it's fine for now